x-pack/plugins/apm/server/routes/data_view/create_static_data_view.ts

```diff
@@ -44,23 +50,46 @@ export async function createStaticDataView({
     }
 
     const apmDataViewTitle = getApmDataViewTitle(setup.indices);
-    const forceOverwrite = await getForceOverwrite({
+    const shouldCreateOrUpdate = await getShouldCreateOrUpdate({
       apmDataViewTitle,
-      savedObjectsClient,
+      dataViewService,
     });
 
+    if (!shouldCreateOrUpdate) {
+      return false;
+    }
+
     try {
-      await withApmSpan('create_index_pattern_saved_object', () =>
-        savedObjectsClient.create(
-          'index-pattern',
-          getApmDataViewAttributes(apmDataViewTitle),
+      await withApmSpan('create_index_pattern_saved_object', async () => {
+        await dataViewService.createAndSave(
           {
+            allowNoIndex: true,
             id: APM_STATIC_DATA_VIEW_ID,
-            overwrite: forceOverwrite,
-            namespace: spaceId,
-          }
-        )
-      );
+            name: 'APM UI',
```

Comment:
This is pretty neat. It's now possible to set a human readable title for a data view. 

Before:

<img src="https://user-images.githubusercontent.com/209966/179908583-662664f7-45ac-4696-bee2-2487007201bd.png" width="250">

After (I've manually added the names for the other Observability apps):

<img src="https://user-images.githubusercontent.com/209966/179908601-29ef06eb-cfa7-428b-8037-4d111d3b7237.png" width="250">